music21/abcFormat/testFiles.py

```diff
@@ -612,8 +676,116 @@ def testBasic(self):
             if title == 'D Fragment':
                 sharps = s.parts[0].keySignature.sharps
                 self.assertEqual(sharps, 7, 'C# key signature should be parsed as 7 sharps')
+                
+    def testAbc21(self):
+        from music21 import abcFormat, note
+        from music21.abcFormat import translate
+        from music21.musicxml import m21ToXml
 
+        af = abcFormat.ABCFile(abcVersion="2.1")
+
+        gex = m21ToXml.GeneralObjectExporter()
         
+        files = [carryThrough]
+        for i, tf in enumerate(files):
+            ah = af.readstr(tf)
+            title = ah.getTitle()
+            environLocal.printDebug([title])
+            s = translate.abcToStreamScore(ah)
+            try:
+                unused_out = gex.parse(s)
+            except UnicodeDecodeError as ude:
+                environLocal.warn('About to fail on ABC file #{}'.format(i))
+                raise ude
+            if title == 'Through Measure':
+                notes = s.flat.getElementsByClass(note.Note)
```

Comment:
wrap `s.flat....` in a `list()` call to make this run faster (just a good practice -- otherwise we're using an expensive iterator)